# Unindo as bases

In [1]:
# Importando os pacotes necessários
import pandas as pd

# Lendo os arquivos .CSV
df_oct = pd.read_csv('2019Oct_SQL.csv')
df_nov = pd.read_csv('2019Nov_SQL.csv')
df_dec = pd.read_csv('2019Dec_SQL.csv')
df_jan = pd.read_csv('2020Jan_SQL.csv')
df_feb = pd.read_csv('2020Feb_SQL.csv')

## OUTUBRO/2019

In [2]:
# Dividindo a coluna 'category_code' em 3 novas colunas
df_oct[['category', 'categ2', 'categ3']] = df_oct['category_code'].str.split('.', expand=True)

In [4]:
# Excluindo as colunas 'category_code', 'categ2' e 'categ3'
df_oct2 = df_oct.drop(columns=['category_code', 'categ2', 'categ3'])

# Reorganizando as colunas
df_oct3 = df_oct2[['Data', 'Hora', 'event_type', 'product_id', 'category_id', 'category', 'brand', 'price', 'user_id', 'user_session']]

# Visualizando as 5 primeiras linhas do 'df_oct3'
df_oct3.head()

Data      Hora event_type  product_id          category_id category  \
0  2019-10-01  00:00:00       cart     5773203  1487580005134238553      NaN   
1  2019-10-01  00:00:03       cart     5773353  1487580005134238553      NaN   
2  2019-10-01  00:00:07       cart     5881589  2151191071051219817      NaN   
3  2019-10-01  00:00:07       cart     5723490  1487580005134238553      NaN   
4  2019-10-01  00:00:15       cart     5881449  1487580013522845895      NaN   

    brand  price    user_id                          user_session  
0  runail   2.62  463240011  26dd6e6e-4dac-4778-8d2c-92e149dab885  
1  runail   2.62  463240011  26dd6e6e-4dac-4778-8d2c-92e149dab885  
2  lovely  13.48  429681830  49e8d843-adf3-428b-a2c3-fe8bc6a307c9  
3  runail   2.62  463240011  26dd6e6e-4dac-4778-8d2c-92e149dab885  
4  lovely   0.56  429681830  49e8d843-adf3-428b-a2c3-fe8bc6a307c9

In [5]:
df_oct3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4102283 entries, 0 to 4102282
Data columns (total 10 columns):
 #   Column        Dtype  
---  ------        -----  
 0   Data          object 
 1   Hora          object 
 2   event_type    object 
 3   product_id    int64  
 4   category_id   int64  
 5   category      object 
 6   brand         object 
 7   price         float64
 8   user_id       int64  
 9   user_session  object 
dtypes: float64(1), int64(3), object(6)
memory usage: 313.0+ MB


In [6]:
# Verifica se há linhas duplicadas
duplicadas = df_oct3.duplicated()

# Conta o número de linhas duplicadas
num_duplicadas = duplicadas.sum()

# Exibe o número de linhas duplicadas
print(f"Existem {num_duplicadas} linhas duplicadas no DataFrame.")

Existem 213155 linhas duplicadas no DataFrame.


In [7]:
# Remove linhas duplicadas
df_oct4 = df_oct3.drop_duplicates()

# Exibe o novo tamanho do DataFrame
print(f"O DataFrame agora tem {df_oct4.shape[0]} linhas após remover duplicatas.")

O DataFrame agora tem 3889128 linhas após remover duplicatas.


## NOVEMBRO/2019

In [9]:
# Dividindo a coluna 'category_code' em 3 novas colunas
df_nov[['category', 'categ2', 'categ3']] = df_nov['category_code'].str.split('.', expand=True)

In [10]:
# Excluindo as colunas 'category_code', 'categ2' e 'categ3'
df_nov2 = df_nov.drop(columns=['category_code', 'categ2', 'categ3'])

# Reorganizando as colunas
df_nov3 = df_nov2[['Data', 'Hora', 'event_type', 'product_id', 'category_id', 'category', 'brand', 'price', 'user_id', 'user_session']]

# Visualizando as 5 primeiras linhas do 'df_nov3'
df_nov3.head()

Data      Hora        event_type  product_id          category_id  \
0  2019-11-01  00:00:02              view     5802432  1487580009286598681   
1  2019-11-01  00:00:09              cart     5844397  1487580006317032337   
2  2019-11-01  00:00:10              view     5837166  1783999064103190764   
3  2019-11-01  00:00:11              cart     5876812  1487580010100293687   
4  2019-11-01  00:00:24  remove_from_cart     5826182  1487580007483048900   

  category     brand  price    user_id                          user_session  
0      NaN       NaN   0.32  562076640  09fafd6c-6c99-46b1-834f-33527f4de241  
1      NaN       NaN   2.38  553329724  2067216c-31b5-455d-a1cc-af0575a34ffb  
2      NaN       pnb  22.22  556138645  57ed222e-a54a-4907-9944-5a875c2d7f4f  
3      NaN  jessnail   3.16  564506666  186c1951-8052-4b37-adce-dd9644b1d5f7  
4      NaN       NaN   3.33  553329724  2067216c-31b5-455d-a1cc-af0575a34ffb

In [11]:
df_nov3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4635837 entries, 0 to 4635836
Data columns (total 10 columns):
 #   Column        Dtype  
---  ------        -----  
 0   Data          object 
 1   Hora          object 
 2   event_type    object 
 3   product_id    int64  
 4   category_id   int64  
 5   category      object 
 6   brand         object 
 7   price         float64
 8   user_id       int64  
 9   user_session  object 
dtypes: float64(1), int64(3), object(6)
memory usage: 353.7+ MB


In [12]:
# Verifica se há linhas duplicadas
duplicadas = df_nov3.duplicated()

# Conta o número de linhas duplicadas
num_duplicadas = duplicadas.sum()

# Exibe o número de linhas duplicadas
print(f"Existem {num_duplicadas} linhas duplicadas no DataFrame.")

Existem 246693 linhas duplicadas no DataFrame.


In [13]:
# Remove linhas duplicadas
df_nov4 = df_nov3.drop_duplicates()

# Exibe o novo tamanho do DataFrame
print(f"O DataFrame agora tem {df_nov4.shape[0]} linhas após remover duplicatas.")

O DataFrame agora tem 4389144 linhas após remover duplicatas.


## DEZEMBRO/2019

In [15]:
# Dividindo a coluna 'category_code' em 3 novas colunas
df_dec[['category', 'categ2', 'categ3']] = df_dec['category_code'].str.split('.', expand=True)

In [16]:
# Excluindo as colunas 'category_code', 'categ2' e 'categ3'
df_dec2 = df_dec.drop(columns=['category_code', 'categ2', 'categ3'])

# Reorganizando as colunas
df_dec3 = df_dec2[['Data', 'Hora', 'event_type', 'product_id', 'category_id', 'category', 'brand', 'price', 'user_id', 'user_session']]

# Visualizando as 5 primeiras linhas do 'df_dec3'
df_dec3.head()

Data      Hora        event_type  product_id          category_id  \
0  2019-12-01  00:00:00  remove_from_cart     5712790  1487580005268456287   
1  2019-12-01  00:00:00              view     5764655  1487580005411062629   
2  2019-12-01  00:00:02              cart        4958  1487580009471148064   
3  2019-12-01  00:00:05              view     5848413  1487580007675986893   
4  2019-12-01  00:00:07              view     5824148  1487580005511725929   

  category      brand  price    user_id                          user_session  
0      NaN      f.o.x   6.27  576802932  51d85cb0-897f-48d2-918b-ad63965c12dc  
1      NaN        cnd  29.05  412120092  8adff31e-2051-4894-9758-224bfa8aec18  
2      NaN     runail   1.19  494077766  c99a50e8-2fac-4c4d-89ec-41c05f114554  
3      NaN  freedecor   0.79  348405118  722ffea5-73c0-4924-8e8f-371ff8031af4  
4      NaN        NaN   5.56  576005683  28172809-7e4a-45ce-bab0-5efa90117cd5

In [17]:
df_dec3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3533286 entries, 0 to 3533285
Data columns (total 10 columns):
 #   Column        Dtype  
---  ------        -----  
 0   Data          object 
 1   Hora          object 
 2   event_type    object 
 3   product_id    int64  
 4   category_id   int64  
 5   category      object 
 6   brand         object 
 7   price         float64
 8   user_id       int64  
 9   user_session  object 
dtypes: float64(1), int64(3), object(6)
memory usage: 269.6+ MB


In [18]:
# Verifica se há linhas duplicadas
duplicadas = df_dec3.duplicated()

# Conta o número de linhas duplicadas
num_duplicadas = duplicadas.sum()

# Exibe o número de linhas duplicadas
print(f"Existem {num_duplicadas} linhas duplicadas no DataFrame.")

Existem 183860 linhas duplicadas no DataFrame.


In [19]:
# Remove linhas duplicadas
df_dec4 = df_dec3.drop_duplicates()

# Exibe o novo tamanho do DataFrame
print(f"O DataFrame agora tem {df_dec4.shape[0]} linhas após remover duplicatas.")

O DataFrame agora tem 3349426 linhas após remover duplicatas.


## JANEIRO/2020

In [20]:
# Dividindo a coluna 'category_code' em 3 novas colunas
df_jan[['category', 'categ2', 'categ3']] = df_jan['category_code'].str.split('.', expand=True)

In [21]:
# Excluindo as colunas 'category_code', 'categ2' e 'categ3'
df_jan2 = df_jan.drop(columns=['category_code', 'categ2', 'categ3'])

# Reorganizando as colunas
df_jan3 = df_jan2[['Data', 'Hora', 'event_type', 'product_id', 'category_id', 'category', 'brand', 'price', 'user_id', 'user_session']]

# Visualizando as 5 primeiras linhas do 'df_jan3'
df_jan3.head()

Data      Hora event_type  product_id          category_id category  \
0  2020-01-01  00:00:00       view     5809910  1602943681873052386      NaN   
1  2020-01-01  00:00:09       view     5812943  1487580012121948301      NaN   
2  2020-01-01  00:00:19       view     5798924  1783999068867920626      NaN   
3  2020-01-01  00:00:24       view     5793052  1487580005754995573      NaN   
4  2020-01-01  00:00:25       view     5899926  2115334439910245200      NaN   

      brand  price    user_id                          user_session  
0   grattol   5.24  595414620  4adb70bb-edbd-4981-b60f-a05bfd32683a  
1  kinetics   3.97  595414640  c8c5205d-be43-4f1d-aa56-4828b8151c8a  
2    zinger   3.97  595412617  46a5010f-bd69-4fbe-a00d-bb17aa7b46f3  
3       NaN   4.92  420652863  546f6af3-a517-4752-a98b-80c4c5860711  
4       NaN   3.92  484071203  cff70ddf-529e-4b0c-a4fc-f43a749c0acb

In [22]:
df_jan3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4264752 entries, 0 to 4264751
Data columns (total 10 columns):
 #   Column        Dtype  
---  ------        -----  
 0   Data          object 
 1   Hora          object 
 2   event_type    object 
 3   product_id    int64  
 4   category_id   int64  
 5   category      object 
 6   brand         object 
 7   price         float64
 8   user_id       int64  
 9   user_session  object 
dtypes: float64(1), int64(3), object(6)
memory usage: 325.4+ MB


In [23]:
# Verifica se há linhas duplicadas
duplicadas = df_jan3.duplicated()

# Conta o número de linhas duplicadas
num_duplicadas = duplicadas.sum()

# Exibe o número de linhas duplicadas
print(f"Existem {num_duplicadas} linhas duplicadas no DataFrame.")

Existem 225100 linhas duplicadas no DataFrame.


In [24]:
# Remove linhas duplicadas
df_jan4 = df_jan3.drop_duplicates()

# Exibe o novo tamanho do DataFrame
print(f"O DataFrame agora tem {df_jan4.shape[0]} linhas após remover duplicatas.")

O DataFrame agora tem 4039652 linhas após remover duplicatas.


## FEVEREIRO/2020

In [25]:
# Dividindo a coluna 'category_code' em 3 novas colunas
df_feb[['category', 'categ2', 'categ3']] = df_feb['category_code'].str.split('.', expand=True)

In [26]:
# Excluindo as colunas 'category_code', 'categ2' e 'categ3'
df_feb2 = df_feb.drop(columns=['category_code', 'categ2', 'categ3'])

# Reorganizando as colunas
df_feb3 = df_feb2[['Data', 'Hora', 'event_type', 'product_id', 'category_id', 'category', 'brand', 'price', 'user_id', 'user_session']]

# Visualizando as 5 primeiras linhas do 'df_feb3'
df_feb3.head()

Data      Hora event_type  product_id          category_id category  \
0  2020-02-01  00:00:01       cart     5844305  1487580006317032337      NaN   
1  2020-02-01  00:00:03       view     5769925  1487580013841613016      NaN   
2  2020-02-01  00:00:08       view     5817765  1487580008246412266      NaN   
3  2020-02-01  00:00:11       view     5877033  1487580010100293687      NaN   
4  2020-02-01  00:00:12       cart     5814871  1487580008112194531      NaN   

    brand  price    user_id                          user_session  
0     NaN   2.14  485174092  4be9643a-420b-4c6b-83dd-a15e772fbf7a  
1  kapous   4.22  594621622  a88baf11-9cd0-4362-bde4-1bfeed3f641d  
2  zeitun  11.03  495404942  3a569c8d-d848-4f09-a925-33f673d84c46  
3    milv   3.49  564814969  7feb39e5-bb7b-4b2b-a546-3e3b7c56326e  
4  zinger   2.54  551205603  106a7c7f-7fa1-4463-8bec-0ea7b02191cb

In [27]:
df_feb3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4156682 entries, 0 to 4156681
Data columns (total 10 columns):
 #   Column        Dtype  
---  ------        -----  
 0   Data          object 
 1   Hora          object 
 2   event_type    object 
 3   product_id    int64  
 4   category_id   int64  
 5   category      object 
 6   brand         object 
 7   price         float64
 8   user_id       int64  
 9   user_session  object 
dtypes: float64(1), int64(3), object(6)
memory usage: 317.1+ MB


In [28]:
# Verifica se há linhas duplicadas
duplicadas = df_feb3.duplicated()

# Conta o número de linhas duplicadas
num_duplicadas = duplicadas.sum()

# Exibe o número de linhas duplicadas
print(f"Existem {num_duplicadas} linhas duplicadas no DataFrame.")

Existem 240290 linhas duplicadas no DataFrame.


In [29]:
# Remove linhas duplicadas
df_feb4 = df_feb3.drop_duplicates()

# Exibe o novo tamanho do DataFrame
print(f"O DataFrame agora tem {df_feb4.shape[0]} linhas após remover duplicatas.")

O DataFrame agora tem 3916392 linhas após remover duplicatas.


# Unindo os dataframes

In [30]:
# Lista de todos os DataFrames
dfs = [df_oct4, df_nov4, df_dec4, df_jan4, df_feb4]

In [31]:
# Concatenar todos os DataFrames em um só
df_unificado = pd.concat(dfs, ignore_index=True)

In [32]:
# Visualizando as 5 primeiras linhas do 'df_unificado'
df_unificado.head(5)

Data      Hora event_type  product_id          category_id category  \
0  2019-10-01  00:00:00       cart     5773203  1487580005134238553      NaN   
1  2019-10-01  00:00:03       cart     5773353  1487580005134238553      NaN   
2  2019-10-01  00:00:07       cart     5881589  2151191071051219817      NaN   
3  2019-10-01  00:00:07       cart     5723490  1487580005134238553      NaN   
4  2019-10-01  00:00:15       cart     5881449  1487580013522845895      NaN   

    brand  price    user_id                          user_session  
0  runail   2.62  463240011  26dd6e6e-4dac-4778-8d2c-92e149dab885  
1  runail   2.62  463240011  26dd6e6e-4dac-4778-8d2c-92e149dab885  
2  lovely  13.48  429681830  49e8d843-adf3-428b-a2c3-fe8bc6a307c9  
3  runail   2.62  463240011  26dd6e6e-4dac-4778-8d2c-92e149dab885  
4  lovely   0.56  429681830  49e8d843-adf3-428b-a2c3-fe8bc6a307c9

In [33]:
# Visualizando as 5 últimas linhas do 'df_unificado'
df_unificado.tail(5)

Data      Hora event_type  product_id          category_id  \
19583737  2020-02-29  23:59:32       view     5885416  1487580005092295511   
19583738  2020-02-29  23:59:39       cart     5550686  1487580008145748965   
19583739  2020-02-29  23:59:45       view     5850628  1602943681873052386   
19583740  2020-02-29  23:59:54       view     5716351  1487580010872045658   
19583741  2020-02-29  23:59:59       view     5927709  1487580013950664926   

         category    brand  price    user_id  \
19583737      NaN  grattol   6.27  622082947   
19583738      NaN      NaN   1.11  459705611   
19583739      NaN  grattol   5.24  622090043   
19583740      NaN    irisk   0.79  619841242   
19583741      NaN   de.lux   3.17  621669991   

                                  user_session  
19583737  fb29909b-6ef5-4662-b4ee-288e73e5dc10  
19583738  05d2add3-01f7-47ee-8364-27341673227f  
19583739  ab7d349f-db5d-4790-8ab1-31e5c894459d  
19583740  18af673b-7fb9-4202-a66d-5c855bc0fd2d  
19583741  daa3c677-5360-4327-847e-d29658716b99

In [35]:
df_unificado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19583742 entries, 0 to 19583741
Data columns (total 10 columns):
 #   Column        Dtype  
---  ------        -----  
 0   Data          object 
 1   Hora          object 
 2   event_type    object 
 3   product_id    int64  
 4   category_id   int64  
 5   category      object 
 6   brand         object 
 7   price         float64
 8   user_id       int64  
 9   user_session  object 
dtypes: float64(1), int64(3), object(6)
memory usage: 1.5+ GB


O dataframe unficado ficou com 10 colunas, 19583742 linhas e um tamanho de aproximadamente 1,5 Gb.

In [36]:
# Salvando o df_unificado num arquivo .CSV
df_unificado.to_csv('fulldb.csv', index=False)